x/gamm/keeper/pool.go

```diff
@@ -15,6 +15,44 @@ import (
 	swaproutertypes "github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
 )
 
+// TODO spec and tests
+func (k Keeper) InitializePool(ctx sdk.Context, pool swaproutertypes.PoolI, creatorAddress sdk.AccAddress) error {
+	poolId := pool.GetId()
+
+	// Add the share token's meta data to the bank keeper.
+	poolShareBaseDenom := types.GetPoolShareDenom(poolId)
+	poolShareDisplayDenom := fmt.Sprintf("GAMM-%d", poolId)
+	k.bankKeeper.SetDenomMetaData(ctx, banktypes.Metadata{
+		Description: fmt.Sprintf("The share token of the gamm pool %d", poolId),
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    poolShareBaseDenom,
+				Exponent: 0,
+				Aliases: []string{
+					"attopoolshare",
```

Comment:
I assume this is copied over right? what does "attopoolshare" mean?

Reply:
Yes, this is copied over. I don't know what it means.

Here's the link to confirm presence on main and 1:1 copy:
https://github.com/osmosis-labs/osmosis/blob/c659f2088a3b0a7b5f3fd33c7d48c834fbfd8581/x/gamm/keeper/pool_service.go#L169

Reply:
In general, I recommend we try to separate out file moves / refactors from logic changes.

`atto` is the SI prefix for `10^(-18)`